### ISSUE TITLE
Unable to save custom API tool

### ISSUE THREAD

**Issue by bowenliang123** [langgenius/dify#2672]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### Dify version

main/0.5.7

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

- Create a custom tool, use the default JSON sample of "Weather"
- Click save button, and then failed
<img width="1499" alt="image" src="https://github.com/langgenius/dify/assets/1935105/2d4c19ca-bfc9-4333-898b-24210860a6f9">

```
fd9d1056-83a0a335d995ca9c.js:1 Uncaught TypeError: Cannot delete property 'api_key_header' of #<Object>
    at onClick (4168-d9f3874f72da8251.js:1:49527)
    at Object.aS (fd9d1056-83a0a335d995ca9c.js:1:71761)
    at aN (fd9d1056-83a0a335d995ca9c.js:1:71915)
    at fd9d1056-83a0a335d995ca9c.js:1:138576
    at sw (fd9d1056-83a0a335d995ca9c.js:1:138675)
    at sS (fd9d1056-83a0a335d995ca9c.js:1:139089)
    at fd9d1056-83a0a335d995ca9c.js:1:145232
    at oK (fd9d1056-83a0a335d995ca9c.js:1:92812)
    at iL (fd9d1056-83a0a335d995ca9c.js:1:122042)
    at sN (fd9d1056-83a0a335d995ca9c.js:1:140397)
```

**Comment by crazywoola** [comment]
Link https://github.com/langgenius/dify/pull/2621

**Comment by bowenliang123** [comment]
> Link #2621

It works fine if #2621 is reverted.
cc @Yeuoly

**Comment by Yeuoly** [comment]
fixed #2673
